pennylane/tape/qscript.py

```diff
@@ -550,18 +551,26 @@ def trainable_params(self, param_indices):
 
         self._trainable_params = sorted(set(param_indices))
 
-    def get_operation(self, idx):
+    def get_operation(self, idx, return_op_index=False):
         """Returns the trainable operation, and the corresponding operation argument
         index, for a specified trainable parameter index.
 
         Args:
             idx (int): the trainable parameter index
-
+            return_op_index (bool): the trainable parameter index
         Returns:
             tuple[.Operation, int]: tuple containing the corresponding
             operation, and an integer representing the argument index,
             for the provided trainable parameter.
         """
+        if not return_op_index:
+            warnings.warn(
+                "The get_operation will soon be updated to also return the index of the trainable operation in the tape."
+                "If you want to switch to the new behavior, you can pass `return_op_index==True`"
+            )
+        else:
+            return self._get_operation(idx)
```

Comment:
swapping the if-else for style, but also only using one equal-sign in the code suggestion given to the user
```suggestion
        if return_op_index:
            return self._get_operation(idx)
        warnings.warn(
            "The get_operation will soon be updated to also return the index of the trainable operation in the tape."
            "If you want to switch to the new behavior, you can pass `return_op_index=True`"
        )
```